src/server/rpc/procedures/chart/chart.js

```diff
@@ -66,23 +66,33 @@ function calcRanges(lines, isCategorical){
 }
 
 function prepareData(input, options) {
-    // if the input is one line convert it to appropriate format
     const xShouldBeNumeric = !options.isCategorical && !options.isTimeSeries;
-
-    if (! Array.isArray(input[0][0])){
+    
+    // if the input is one line convert it to appropriate format
+    if (Array.isArray(input) && !Array.isArray(input[0]) || !Array.isArray(input[0][0]) && input[0].length === 2){
```

Comment:
the main changes for this PR happen here and a few lines down where you check for 
>> // If only one dimension is given
and the rest are formatting updates correct?

Reply:
Yeah, that's the only major change.